fdbserver/MoveKeys.actor.cpp

```diff
@@ -163,6 +288,86 @@ ACTOR Future<Optional<UID>> checkReadWrite(Future<ErrorOr<GetShardStateReply>> f
 	return Optional<UID>(uid);
 }
 
+// Cleans up dest servers of a single shard, and unassigns the keyrange from the dest servers if necessary.
+ACTOR Future<Void> cleanUpSingleShardDataMove(Database occ,
+                                              KeyRange keys,
+                                              MoveKeysLock lock,
+                                              FlowLock* cleanUpDataMoveParallelismLock,
+                                              UID dataMoveId,
```

Comment:
Based on the code, we're actually cleaning up the data move referred by the key range. Does this function guarantee only one shard will be cleaned up?

Reply:
Yes, this function is used for handle the 'legacy' data moves scheduled w/o `dataMoveId`, e.g., when shard-based-move is enabled, there could be data moves in progress, and this function will be used to clean up those data moves in the old format.